src/main/scala/wdl4s/MemorySize.scala

```diff
@@ -0,0 +1,39 @@
+package wdl4s
+
+import wdl4s.parser.MemoryUnit
+
+import scala.language.postfixOps
+import scala.util.{Failure, Success, Try}
+import scalaz.Scalaz._
+
+object MemorySize {
+  val memoryPattern = """(\d+(?:\.\d+)?)\s*(\w+)""".r
+  def parse(unparsed: String): Try[MemorySize] = {
```

Comment:
I would take that a step further and opine that a library should be throwing an exception and letting the consumer decide how to handle it. wdl4s doesn't currently do that in many places but that's an artifact of its heritage